dmscripts/models/queries.py

```diff
@@ -42,6 +42,15 @@ def process_fields(rules, data):
     return data
 
 
+def rename_fields(rules, data):
+    """ Return data after renaming any keys using the renaming rules """
+    for key in data.keys():
+        if key in rules.keys():
+            data[rules[key]] = data.pop(key)
+
+    return data
+
```

Comment:
So the above function does a couple of things that I'd be sure to note, if not be wary of.

First things first I'm not sure about the name `rules` for the `rules` arg. When I read it I sort of expected a dict like:
`{'string_field_name': some_method_to_transform_said_field}`
maybe something like:
`rules = {
'name': str.capitalize
'age': lambda i: 'You are %d years old' % i
}`
The function is applying a mapping to the existing field names which is a rule in itself i.e. _The rule_ is if the field is in the mapping dict then swap it with the value in that mapping. Or _apply this mapping to these fields_.

Second thing, you might already know this, with dicts/ DataFrames you don't need to call keys when iterating over them. Dicts/ DataFrames automatically iterate over their keys. 
`assert [i for i in data.keys()] == [i for i in data] == [i for i in data.__iter__()]`
It's a bit of a weird one tbh because it's a part of Python that doesn't feel very intuitive.

Watch out for the whole dict mutability thing. It always feels a bit weird modifying them in place. Although you're returning, and reassigning to, `data` when you use this function below you can just as easily not bother.
`rename_fields(rules, data)`
Is going to do the same as
`data = rename_fields(rules, data)`

That's why `list.extend` and `dict.update` return None

I have a funny feeling that changing the key names/ rebuilding the dataframe from a new dict might mess with the ordering.

In fact scrap that, I believe this is what you want :)
http://stackoverflow.com/a/11354850